pytorch_lightning/logging/tensorboard.py

```diff
@@ -59,23 +59,33 @@ def experiment(self):
     def log_hyperparams(self, params):
         if parse_version(torch.__version__) < parse_version("1.3.0"):
             warn(
-                f"Hyperparameter logging is not available for Torch version {torch.__version__}. "
-                "Skipping log_hyperparams. Upgrade to Torch 1.3.0 or above to enable "
-                "hyperparameter logging"
+                f"Hyperparameter logging is not available for Torch version {torch.__version__}."
+                " Skipping log_hyperparams. Upgrade to Torch 1.3.0 or above to enable"
+                " hyperparameter logging."
             )
             return
-        self.experiment.add_hparams(hparam_dict=vars(params))
+        try:
+            # in case converting from namespace, todo: rather test if it is namespace
+            params = vars(params)
+        except TypeError:
+            pass
+        if params is not None:
+            self.experiment.add_hparams(hparam_dict=dict(params))
```

Comment:
What problem is this fixing? The current implementation is consistent with all the other loggers.

Reply:
a dictionary cannot be converted via `vars`

Reply:
Just for future reference, do we pass a dict anywhere? I thought we always passed some form of Namespace.

Reply:
In almost all tests... =)